File: react-app/src/app/models/tvShow.ts
import { Actor } from "./actor";
import { Director } from "./director";
import { Genre } from "./genre";
import { User } from "./user";

export interface TvShow{
    tvShowId: string;
    title: string;
    description: string;   
    releaseYear: string;
    rating: number;
    seasons:number;
    episodes:number;
    episodeLength:number;
    image?: string;
    imageCarousel?: string;
    genres?: Genre[];
    actors?: Actor[];
    users?: User[];
    directorId: string;
    director?: Director;
}

File: Models/TvShow.cs
namespace ProjWebProgramming.Models
{
    public class TvShow
    {
        public Guid TvShowId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime ReleaseYear { get; set; }
        public decimal Rating { get; set; }
        public int Seasons { get; set; }
        public int Episodes { get; set; }
        public decimal EpisodeLength { get; set; }
        public byte[]? Image { get; set; }
        public byte[]? ImageCarousel { get; set; }
        public ICollection<Genre> Genres { get; set; } = new List<Genre>();
        public List<TvShowGenre>? TvShowGenres { get; set; }
        public ICollection<Actor> Actors { get; set; } = new List<Actor>();
        public List<TvShowActor>? TvShowActors { get; set; }
        public Guid DirectorId { get; set; }
        public Director? Director { get; set; }
        public ICollection<User> Users { get; set; } = new List<User>();
        public List<TvShowWishlist>? TvShowWishlists { get; set; }
    }
}
